.github/workflows/bundleTest-thor.yml

```diff
@@ -0,0 +1,238 @@
+                  echo "uploadArtifact=true" >> $GITHUB_OUTPUT
+              fi
+              # Rename result log file to name of the bundle
+              logname=$(basename $logfilename)
+              bundlelogfilename=${logname//thor/$BUNDLE}
+              printf "%s, %s\n" "$logname" "$bundlelogfilename"
+              mv -v $logfilename /home/runner/HPCCSystems-regression/log/ml-$bundlelogfilename
+          
+          }
+
+          BUNDLES_TO_TEST=( "ML_Core" "PBblas" "GLM" "DBSCAN" "GNN" "LearningTrees" "TextVectors" "KMeans" "SupportVectorMachines" "LinearRegression" "LogisticRegression" )        
+          while read bundle
+          do
+              bundleRunPath=${bundle%/ecl}                         # remove '/ecl' from the end of the $bundle
+              bundlePath=${bundleRunPath%/OBTTests};       # remove '/OBTTests' from the end of the $bundleRunPath if exists
+              bundleName=${bundlePath%/test}                    # remove '/test' from the end of the $bundlePath if exists
+              bundleName=$(basename $bundleName )         # remove path from $bundleName
+              
+              if [[ "$bundle" =~ "LearningTrees" ]]
+              then
+                  # add a warning supression parameter in the file
+                  for file in "RegressionTestModified.ecl" "ClassificationTestModified.ecl"
+                  do
+                      if [[ $( egrep -c '#ONWARNING\(30004' $bundle/$file ) -eq 0 ]]
+                      then
+                          pushd $bundle 
+                          cp -fv $file $file-back
+                          # Insert a comment and the "#ONWARNING" after the Copyright header
+                          sed -i '/## \*\//a \\n// Patched by the bundleTest on '"$( date '+%Y.%m.%d %H:%M:%S')"' \n#ONWARNING(30004, ignore); // Do not report execute time skew warning' $file
+                          popd
+                      fi
+                  done
+              fi
+              
+              if [[ ! "${BUNDLES_TO_TEST[*]}" =~ "$bundleName"  ]]
+              then
+                  continue
+              fi
+              
+              pushd $bundleRunPath
+          
+              /opt/HPCCSystems/testing/regress/ecl-test run -t thor --config /opt/HPCCSystems/testing/regress/ecl-test.json --timeout 3600 -fthorConnectTimeout=3600 --pq 1 --ef ClassicTestModified.ecl,SVCTest.ecl,ClassificationTestModified.ecl
+              
+              retCode=$( echo $? )
+          
+              if [ ${retCode} -eq 0 ] 
+              then
+                  ProcessLog "$bundleName" "thor"
+              fi
+            
+              popd
+  
+          done< <(find . -iname 'ecl' -type d | sort )
+      - name: Generate ZAP files
+        if: ${{ ! inputs.Generate-Zap == '' }} 
+        run: |  
+          IFS=' ' read -a ML_GENERATE_ZAP_FOR <<< ${{ inputs.Generate-Zap }}
```

Comment:
If any ZAP file is forced to generate, then they should be add to the artifact and upload. Otherwise they are lost at the end of runner execution and no reason to generate them.